src/foam/graphics/TreeGraph.js

```diff
@@ -33,306 +36,76 @@
+           this.time = Date.now();
          }
-       ],
-
-       listeners: [
-        {
-          name: 'doLayout',
-          isFramed: true,
-          documentation: 'Animate layout until positions stabilize',
-          code: function() {
-            var needsLayout = false;
-            // Scale and translate the view to fit in the available window
-            var gw = this.graph.width-110;
-            var w  = this.maxRight - this.maxLeft + 55;
-
-            var x = (-this.maxLeft+25)/w * gw + 55;
-            needsLayout = this.convergeTo(this.x$, x) || needsLayout;
-            
-            if ( this.layout() || needsLayout ) {
-              this.doLayout();
-            }
-            else {
-              this.graph.updateCWidth();
-            }
-          
-            this.graph.invalidate();
+          if ( this.root && this.root.layout() ) {
+            this.invalidate();
+            this.doLayout();
+          } else {
+            // console.log(Date.now() - this.time)
+            this.time = null;
+            this.updateCSize();
           }
         }
-       ]
-     }
-   ],
-
-   listeners: [
+     },
      {
-       name: 'updateCWidth',
+       name: 'updateCSize',
        isFramed: true,
        code: function() {
-        const maxes = {
-          maxLeft: 0,
-          maxRight: 0
-        };
-          
-        function traverseAndCompare(root) {
-          if ( root.maxLeft < maxes.maxLeft ) maxes.maxLeft = root.maxLeft;
-          if ( root.maxRight > maxes.maxRight ) maxes.maxRight = root.maxRight;
+         const maxes  = {
+           maxLeft: Number.MAX_SAFE_INTEGER,
+           maxRight: Number.MIN_SAFE_INTEGER
+         }
 
-          for ( var i = 0; i < root.children.length; i++ ){
-            traverseAndCompare(root.children[i]);
-          }
-        }
+         this.root.outline.forEach(level => {
+           maxes.maxLeft = Math.min(level.left, maxes.maxLeft);
+           maxes.maxRight = Math.max(level.right, maxes.maxRight);
+         })
 
-        traverseAndCompare(this.root);
+         var width = Math.abs(maxes.maxLeft - maxes.maxRight);
+         var height = (this.nodeHeight * 2) * (this.root.outline.length + 1);
 
-        // needed to use the adjustments to width in order to account for proper fitting on the screen
-        // since by default with out it, the leftmost node and right most node get cutoff by half
-        // padding adjustments are there to proper spacing and also for the edge connectors to 
-        // fully render
-        var width = Math.abs(maxes.maxLeft - maxes.maxRight) + this.nodeWidth + this.padding * 4;
-        var delta = Math.abs(this.width - width) / width;
+         var deltaX = Math.abs(this.width - width) / width;
+         var deltaY = Math.abs(this.height - height) / height;
 
-        if ( delta > 0.01 ) {
-          this.width = width;
-          this.doLayout();
-        }
-       }
+         if ( deltaX > 0.01 || deltaY > 0.01 ) {
```

Comment:
Think this was from the legacy code where we wanted to reduce the amount this.doLayout() was called to have the graph render faster by having a delta rather than convergence on the exact value. Changed it to `this.width != width || this.height` since the new graph renders super fast anyways